spring/src/main/java/io/kubernetes/client/spring/extended/controller/config/KubernetesInformerAutoConfiguration.java

```diff
@@ -25,10 +28,32 @@
 @ConditionalOnKubernetesInformerEnabled
 public class KubernetesInformerAutoConfiguration {
 
+  private static final Log LOG = LogFactory.getLog(KubernetesInformerAutoConfiguration.class);
+
   @Bean
   @ConditionalOnMissingBean
   public ApiClient defaultApiClient() throws IOException {
-    return ClientBuilder.defaultClient();
+    try {
```

Comment:
The current [`defaultClient()`](https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/util/ClientBuilder.java#L103) tries to load from a local config in the following order:
* `$KUBECONFIG` if it exists
* `$HOME/.kube/config` if it exists
* in-cluster configuration if it exists
* localhost:8080 if all else fails

I think we should maintain that order of selection.

What are the specific exceptions that you are seeing? They're not listed in the issue.

I'd prefer we fixed the specific exceptions while maintaining the same load order.

Reply:
And perhaps it is better to add the exception handling to the `defaultClient()` function instead of the Spring loader.